compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -79,10 +88,18 @@ object ProtoTypes {
       val savedConstraint = ctx.typerState.constraint
       val res = pt.widenExpr match {
         case pt: FunProto =>
-          mt match {
-            case mt: MethodType => constrainResult(resultTypeApprox(mt), pt.resultType)
+          mt match
+            case mt: MethodType =>
+              constrainResult(resultTypeApprox(mt), pt.resultType)
+              && {
+                if pt.constrainResultDeep
+                   && mt.isImplicitMethod == (pt.applyKind == ApplyKind.Using)
+                then
+                  val tpargs = pt.args.lazyZip(mt.paramInfos).map(pt.typedArg)
+                  tpargs.tpes.corresponds(mt.paramInfos)(_ <:< _)
```

Comment:
Looks like this could be something like:
```suggestion
                  pt.args.lazyZip(mt.paramInfos).forall((arg, paramInfo) =>
                    pt.typedArg(arg).tpe <:< paramInfo)
```
Or does running `typedArg` on all the arguments first makes a difference? If so that's worth a comment I think.

Reply:
yes, i think that change is better